contracts/plugins/assets/curve/CurveRecursiveCollateral.sol

```diff
@@ -0,0 +1,192 @@
+
+                // Save prices if priced
+                if (high < FIX_MAX) {
+                    savedLowPrice = low;
+                    savedHighPrice = high;
+                    lastSave = uint48(block.timestamp);
+                } else {
+                    // must be unpriced
+                    // untested:
+                    //      validated in other plugins, cost to test here is high
+                    assert(low == 0);
+                }
+
+                // If the price is below the default-threshold price, default eventually
+                // uint192(+/-) is the same as Fix.plus/minus
+                if (low == 0 || _anyDepeggedInPool() || _anyDepeggedOutsidePool()) {
+                    markStatus(CollateralStatus.IFFY);
+                } else {
+                    markStatus(CollateralStatus.SOUND);
+                }
+            } catch (bytes memory errData) {
+                // see: docs/solidity-style.md#Catching-Empty-Data
+                if (errData.length == 0) revert(); // solhint-disable-line reason-string
+                markStatus(CollateralStatus.IFFY);
+            }
+        } catch (bytes memory errData) {
+            // see: docs/solidity-style.md#Catching-Empty-Data
+            if (errData.length == 0) revert(); // solhint-disable-line reason-string
+            markStatus(CollateralStatus.DISABLED);
+        }
+
+        CollateralStatus newStatus = status();
+        if (oldStatus != newStatus) {
+            emit CollateralStatusChanged(oldStatus, newStatus);
+        }
+    }
+
+    /// @dev Not up-only! The RToken can devalue its exchange rate peg
+    /// @return {ref/tok} Quantity of whole reference units per whole collateral tokens
+    function underlyingRefPerTok() public view virtual override returns (uint192) {
+        // {ref/tok} = quantity of the reference unit token in the pool per vault token
+        // the vault is 1:1 with the LP token
+
+        // {lpToken@t=0/lpToken}
+        uint192 virtualPrice = _safeWrap(curvePool.get_virtual_price());
+        // this is missing the fact that USDC+ has also appreciated in this time
+
+        // {BU/rTok}
+        uint192 rTokenRate = divuu(rToken.basketsNeeded(), rToken.totalSupply());
+        // not worth the gas to protect against div-by-zero
+
+        // The rTokenRate is not up-only! We should expect decreases when other
+        // collateral default and there is not enough RSR stake to cover the hole.
+
+        // {ref/tok} = {ref/lpToken} = {lpToken@t=0/lpToken} * {1} * 2{ref/lpToken@t=0}
+        return virtualPrice.mul(rTokenRate.sqrt()).mulu(2); // LP token worth twice as much
+    }
+
+    // === Internal ===
+
+    // Override this later to implement non-standard recursive pools
+    function _anyDepeggedInPool() internal view virtual override returns (bool) {
+        // Check reference token
+        try this.tokenPrice(0) returns (uint192 low, uint192 high) {
```

Comment:
Probably needs a comment saying that this only works where the first token is the paired token and the second token is the RToken.